express-notes-api/index.js

```diff
@@ -0,0 +1,89 @@
+const express = require('express');
+const fs = require('fs');
+const data = require('./data');
+const app = express();
+
+app.use(express.json());
+
+const saveData = () => {
+  const dataJSON = JSON.stringify(data, null, 2);
+  fs.writeFile('data.json', dataJSON, err => {
+    if (err) {
+      return false;
+    } else {
+      return true;
+    }
+  });
```

Comment:
This doesn't work as expected. `fs.writeFile` is asynchronous (an also returns `undefined`). So if you use a helper function, you need to make it take a callback.

```js
const saveDate = done => {
  const dataJSON = JSON.stringify(data, null, 2);
  fs.writeFile('data.json', dataJSON, done);
};
```

then when you call it:

```js
app.post('/api/notes', (req, res) => {
  if (!req.body.content) {
    res.status(400).json({ error: 'content is a required field' });
  } else {
    data.notes[data.nextId] = { id: data.nextId, content: req.body.content };
    data.nextId++;
    saveData(err => {
      if (err) {
        console.error(err);
        res.status(500).json({ error: 'An unexpected error occurred.' });
        return;
      }
      res.status(201).json(req.body);
    });
});
```

alternatively, you can ditch the `saveData` function and just do your `fs.writeFile` calls directly where you need them. but the key thing is that you put code that runs _after_ `writeFile` in its callback.